comp/core/configsync/configsyncimpl/sync.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package configsyncimpl
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"time"
+
+	apiutil "github.com/DataDog/datadog-agent/pkg/api/util"
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+)
+
+func (cs *configSync) runWithInterval(refreshInterval time.Duration) {
+	ticker := time.NewTicker(refreshInterval)
+	defer ticker.Stop()
+
+	cs.runWithChan(ticker.C)
+}
+
+func (cs *configSync) runWithChan(ch <-chan time.Time) {
+	// whether we managed to contact the core-agent, used to avoid spamming logs
+	connected := true
+	url := cs.url.String()
+
+	cs.Log.Infof("Starting to sync config with core agent at %s", cs.url)
+
+	for {
+		select {
+		case <-cs.ctx.Done():
+			return
+		case <-ch:
+			cfg, err := fetchConfig(cs.ctx, cs.client, cs.Authtoken.Get(), url)
+			if err != nil {
+				if connected {
+					cs.Log.Warnf("Failed to fetch config from core agent: %v", err)
+					connected = false
+				} else {
+					cs.Log.Debugf("Failed to fetch config from core agent: %v", err)
+				}
+				continue
+			}
+
+			cs.Log.Debug("Succeeded to fetch config from core agent")
+			connected = true
```

Comment:
Should we log something at Info level when we recover connection to the agent core ?